openedx/core/djangoapps/ace_common/templates/ace_common/edx_ace/common/base_body.html

```diff
@@ -20,9 +20,10 @@
     {% block preview_text %}{% endblock %}
 </div>
 
-{# Note {beacon_src} is not a template variable that is evaluated by the Django template engine. It is evaluated by #}
-{# Sailthru when the email is sent. Other email providers would need to replace this variable in the HTML as well. #}
-<img src="{beacon_src}" alt="" role="presentation" aria-hidden="true" />
+{# Note {beacon_html} is not a template variable that is evaluated by the Django template engine. #}
```

Comment:
I **like** how you separated out the Sailthru-specific template values into their own Sailthru template files.  

Do you agree @mulby?

Reply:
I also like the separation, however, I wonder if we couldn't delegate some of this to the channel more directly?

That is, when we render the template, pass in the channel that it is being rendered for it, and allow the template to customize itself based on destination channel. I think this would clean up a bunch of stuff downstream.

For example, the beacon handling could look something like this:

```{{ destination_channel.beacon_html | safe }}```

Then, in the sailthru channel, we could simply implement an attribute that returns the appropriate HTML:

```
@property
def beacon_html(self):
    # Note {beacon_src} is not a template variable that is evaluated by the Django template engine. It is evaluated by ...
    return '<img src="{beacon_src}" alt="" role="presentation" aria-hidden="true" />'
```

Similarly, for the action links:

```
                         <!-- Actions -->
                         <td style="padding-bottom: 20px;">
                             {# Note that these variables are evaluated by Sailthru, not the Django template engine #}
                             {% for action_link in destination_channel.action_links %}
                             <p>
                                 <a href="{{ action_link.url }}" style="color: #005686">
                                     {# Note that the action link text must be translated before being inserted here #}
                                     <font color="#005686"><b>{{ action_link.text }}</b></font>
                                 </a>
                             </p>
                             {% endfor %}
                         </td>
```

Reply:
@mulby Which delivery channel ultimately gets used is eventually determined by ACE in policy.py and delivery.py.  And beacon and action-links are channel-type specific.  For example, push notifications may not require them.

Does it make sense to push the `beacon_html` files into edx-ace instead of keeping them in edx-platform?  I don't imagine different IDAs needing their own versions.

And for the issue of having password-reset emails use Django-email instead of Sailthru, we can introduce a `Policy` implementation that requires Django-email over Sailthru for transactional-type emails.

Reply:
Another challenge we are going to have is that right now we only allow a single channel implementation for each type to be enabled at once. ACE will throw an error if there are multiple enabled email channels.

See this check: https://github.com/edx/edx-ace/blob/master/edx_ace/channel/__init__.py#L94